### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by error-four-o-four**
Hey there,

just another small update which consists of tidying up the code a bit and adding a separate `tsconfig.build.json` to exclude the tests folder from the build process. The latter change allows `rollup-plugin-clean` to be removed in order to decrease the package size and speed up the build process slightly.

* all tests pass
* format script has been  executed

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -68,56 +70,53 @@ const conditionMatch = (
...
         const next = key[j + 1];
+        // '**' is invalid
         if (next === '*') break;
-        const pathMatchIdx = findPathMatchIdx(next, i + 1);
-        if (pathMatchIdx === -1) break;
-        data.push(path.slice(i, pathMatchIdx));
+        // get index of next char
+        const k = !next ? pathLen : path.indexOf(next, i + 1);
+        // exit early
+        if (k === -1) break;
```
Comment (error-four-o-four) [inline src/index.ts:100]: Maybe another guard check is required to resolve subpath mappings which have extensions
e.g. './foo/*.js' => './foo/*.ts'

**Comment by error-four-o-four** [comment]
I have some other suggestion but I'd like to ask for your opinion first.

How much do you insist on using `aidly` as a dependency? I think it could be replaced with self-written utilities. This change would also decrease the package size.
On the other hand I'd like to introduce `type-fest` as a dependency because it provides types for 'package.json' files i.e [exports](https://github.com/sindresorhus/type-fest/blob/fee4e04a6facc94457f47901234448901cebf6a7/source/package-json.d.ts#L214C17-L214C18)

I'd also like to restructure the codebase a bit and split it into several modules e.g.
```css
src/
├── __tests__/
│   ├── ..
│   └── ..
├── core/
│   ├── matchers.ts
│   └── index.ts
├── utils/
│   ├── parser.ts
│   ├── validator.ts
│   └── ...
├── types.ts
└── index.ts
```

Last but not least I wonder if you'd be interested in using [pkgroll](https://github.com/privatenumber/pkgroll) as a bundler? It's a pretty straight forward bundler because it bundles the files depending on the entry points defined in 'package.json'.
Otherwise I could also imagine to use `vite` and `vitest` bc the former could also be used to build the github page.

**Comment by imtaotao** [comment]
> 我还有其他建议，但我想先征求您的意见。
> 
> 您坚持使用`aidly`依赖项的程度有多高？我认为可以用自己编写的实用程序代替它。这种变化还会减小包的大小。 另一方面，我想将其作为依赖项引入，因为它为“package.json”文件（即[导出](https://github.com/sindresorhus/type-fest/blob/fee4e04a6facc94457f47901234448901cebf6a7/source/package-json.d.ts#L214C17-L214C18)`type-fest`）提供了类型[](https://github.com/sindresorhus/type-fest/blob/fee4e04a6facc94457f47901234448901cebf6a7/source/package-json.d.ts#L214C17-L214C18)
> 
> 我还想稍微重组一下代码库，并将其分成几个模块，例如
> 
> ```css
> src/
> ├── __tests__/
> │   ├── ..
> │   └── ..
> ├── core/
> │   ├── matchers.ts
> │   └── index.ts
> ├── utils/
> │   ├── parser.ts
> │   ├── validator.ts
> │   └── ...
> ├── types.ts
> └── index.ts
> ```
> 
> 最后但并非最不重要的一点是，我想知道您是否有兴趣使用[pkgroll](https://github.com/privatenumber/pkgroll)作为捆绑器？这是一个相当直接的捆绑器，因为它根据“package.json”中定义的入口点捆绑文件。 否则我也可以想象使用`vite`和`vitest`bc前者也可以用于构建 github 页面。

**Comment by imtaotao** [comment]
1. I think using `Rollup` for bundling is a good option because the output is very clean and the tree-shaking is excellent. `Aidly` is just a pure utility library I wrote, and thanks to Rollup, we only bundle the code that is actually used.
2. I usually use `Vite` to bundle my web projects. As for unit testing, the APIs we use should be seamlessly compatible with both `Jest` and `Vitest`.
3. `Pkgroll` looks great and can reduce our bundling configuration. If you have time, you might want to give it a try.

**Comment by imtaotao** [comment]
The code base structure reorganization is possible, and there may be more other functional support in the future

**Comment by error-four-o-four** [comment]
I'm waiting for some PRs being merged in 'pkgroll'. Afterwards I'll give it a shot ✌️

**Pull request merged**